Libraries/Opc.Ua.Configuration/ApplicationConfigurationBuilder.cs

```diff
@@ -0,0 +1,880 @@
+                        return pkiRoot + "/trustedHttps";
+                    case TrustlistType.IssuerHttps:
+                        return pkiRoot + "/issuerHttps";
+                    case TrustlistType.TrustedUser:
+                        return pkiRoot + "/trustedUser";
+                    case TrustlistType.IssuerUser:
+                        return pkiRoot + "/issuerUser";
+                    case TrustlistType.Rejected:
+                        return pkiRoot + "/rejected";
+                }
+            }
+            else if (pkiRootType.Equals(CertificateStoreType.X509Store, StringComparison.OrdinalIgnoreCase))
+            {
+                switch (trustListType)
+                {
+                    case TrustlistType.Application:
+#if !NETFRAMEWORK
+                        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows) &&
+                            pkiRoot.StartsWith(CertificateStoreIdentifier.CurrentUser, StringComparison.OrdinalIgnoreCase))
+                        {
+                            return pkiRoot + "/My";
+                        }
+#endif
+                        return pkiRoot + "UA_MachineDefault";
+                    case TrustlistType.Trusted:
+                        return pkiRoot + "UA_Trusted";
+                    case TrustlistType.Issuer:
+                        return pkiRoot + "UA_Issuer";
+                    case TrustlistType.TrustedHttps:
+                        return pkiRoot + "UA_Trusted_Https";
+                    case TrustlistType.IssuerHttps:
+                        return pkiRoot + "UA_Issuer_Https";
+                    case TrustlistType.TrustedUser:
+                        return pkiRoot + "UA_Trusted_User";
+                    case TrustlistType.IssuerUser:
+                        return pkiRoot + "UA_Issuer_User";
+                    case TrustlistType.Rejected:
+                        return pkiRoot + "UA_Rejected";
+                }
+            }
+            throw new NotSupportedException("Unsupported store type.");
+        }
+
+        /// <summary>
+        /// Add specified groups of security policies and security modes.
+        /// </summary>
+        /// <param name="includeSign"></param>
+        /// <param name="deprecated"></param>
+        /// <param name="policyNone"></param>
+        private void AddSecurityPolicies(bool includeSign = false, bool deprecated = false, bool policyNone = false)
+        {
+            // create list of supported policies
+            string[] defaultPolicyUris = SecurityPolicies.GetDefaultUris();
+            if (deprecated)
+            {
+                var names = SecurityPolicies.GetDisplayNames();
+                var deprecatedPolicyList = new List<string>();
+                foreach (var name in names)
+                {
+                    var uri = SecurityPolicies.GetUri(name);
+                    if (uri != null)
+                    {
+                        deprecatedPolicyList.Add(uri);
```

Comment:
_deprecatedPolicyList_ is not added to main list.
AddSecurityPolicies()  is called internally only with _deprecated = false_

Reply:
yes, intentionally added a AddPolicy instead of allowing to add all deprecated policies. However the function might be useful for testing, so I keep it around for a bit,

Reply:
updated API design --> I removed addDeprecated in favor of AddPolicy, to avoid to make it too easy to have unsecure policy settings.